src/middlewared/middlewared/plugins/service_/services/incus.py

```diff
@@ -0,0 +1,37 @@
+import os
+import re
+import signal
+
+from pystemd.systemd1 import Unit
+
+from middlewared.plugins.service_.services.base import SimpleService
+
+
+RE_DNSMASQ_PID = re.compile(r'^pid: (\d+)', flags=re.M)
+
+
+class IncusService(SimpleService):
+    name = "incus"
+
+    etc = ["subids"]
+    systemd_unit = "incus"
+
+    async def stop(self):
+        await self._unit_action("Stop")
+        # incus.socket needs to be stopped in addition to the service
+        unit = Unit("incus.socket")
+        unit.load()
+        await self._unit_action("Stop", unit=unit)
+
+        # Incus will run dnsmasq for its managed network and not stop it
+        # when the service is stopped.
+        dnsmasq_pid = '/var/lib/incus/networks/incusbr0/dnsmasq.pid'
```

Comment:
This is a rather strange set of logic. I'd rather just put this into a synchronous method and do `await self.middleware.run_in_thread(_stop_dnsmasq)` or whatever you want to call it.